poetry/utils/shell.py

```diff
@@ -41,7 +42,16 @@ def get(cls):  # type: () -> Shell
 
         try:
             name, path = detect_shell(os.getpid())
-        except (RuntimeError, ShellDetectionFailure):
+        except ShellDetectionFailure:
+            if os.name == "posix":
+                shell = os.environ["SHELL"]
+            elif os.name == "nt":
+                shell = os.environ["COMSPEC"]
+            else:
+                raise RuntimeError
+
+            name, path = Path(shell).stem, shell
+        except RuntimeError:
```

Comment:
Did you intend to catch the inner `RuntimeError` here to add the message to it? Because it won't. Try the following test.py:

```python
class FirstException(Exception):
    pass


class SecondException(Exception):
    pass


try:
    raise FirstException()
except FirstException:
    raise SecondException()
except SecondException:
    print("Caught")
```

Result:

```text
$ python3 test.py 
Traceback (most recent call last):
  File "test.py", line 10, in <module>
    raise FirstException()
__main__.FirstException

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "test.py", line 12, in <module>
    raise SecondException()
__main__.SecondException
```

Reply:
I stopped counting how often I made this mistake :smile: You are absolutely right.